components/core/src/clp_s/CommandLineArguments.hpp

```diff
@@ -184,6 +186,7 @@ class CommandLineArguments {
     size_t m_target_ordered_chunk_size{};
     size_t m_minimum_table_size{1ULL * 1024 * 1024};  // 1 MB
     bool m_disable_log_order{false};
+    std::string m_file_type{"Json"};
```

Comment:
Let's change m_file_type and the return type of get_file_type to an enum.

That way code that calls get_file_type doesn't need to worry about edge cases where the file type doesn't match any known types since the validation already happens inside CommandLineArguments.

You can see what we do with OutputHandlerType for reference.